hibike/virtual_device.py

```diff
@@ -0,0 +1,114 @@
+import struct
```

Comment:
Not reviewing this file now, pending addressing the issues brought up in the last PR.